Saber/DepthBuffer.cpp

```diff
@@ -63,35 +64,52 @@ void DepthBuffer::Resize(
 	m_pDepthBuffer->CreateShaderResourceView(
 		pDevice, m_pSrvsRange->GetNextCpuHandle(), &srvDesc
 	);
-	ThrowIfFailed(pDevice->GetDeviceRemovedReason());
 
-	// Hierarchical Z-Buffer
+	ResizeHZB(pDevice, pAllocator, width, height);
+}
+
+void DepthBuffer::ResizeHZB(Microsoft::WRL::ComPtr<ID3D12Device2> pDevice, Microsoft::WRL::ComPtr<D3D12MA::Allocator> pAllocator, UINT64 width, UINT height) {
+	if (!m_pSinglePassDownsampler) {
+		return;
```

Comment:
if you use early return, maybe return code error or at least true  / false on function toleave possibility process success/ fail?

Reply:
fixed, now using bool return value, also added for >4096 resolution